src/MoBi.Presentation/Tasks/TableFormulaTask.cs

```diff
@@ -64,5 +89,50 @@ public IMoBiCommand SetRestartSolver(TableFormula formula, ValuePoint valuePoint
       {
          return new SetRestartSolverInValuePointCommand(formula, valuePoint, newRestartSolverValue, buildingBlock);
       }
+
+      public TableFormula ImportTableFormula()
+      {
+         var dataImporterSettings = new DataImporterSettings
+         {
+            Caption = $"{AppConstants.PRODUCT_NAME_WITH_TRADEMARK} - {AppConstants.Captions.ImportTableFormula}",
+            IconName = ApplicationIcons.Formula.IconName
+         };
+         dataImporterSettings.AddNamingPatternMetaData("File");
+
+         var importedFormula = _dataImporter.ImportDataSets(
+            new List<MetaDataCategory>(),
+            _dataImporter.ColumnInfosForObservedData(),
+            dataImporterSettings,
+            _dialogCreator.AskForFileToOpen(Captions.Importer.OpenFile, Captions.Importer.ImportFileFilter, DirectoryKey.OBSERVED_DATA)
+         ).DataRepositories.FirstOrDefault();
+         return importedFormula == null ? null : formulaFrom(importedFormula);
+      }
+
+      private TableFormula formulaFrom(DataRepository dataRepository)
+      {
+         var baseGrid = dataRepository.BaseGrid;
+         var valueColumn = dataRepository.AllButBaseGrid().Single();
+         var formula = newTableFormula();
+
+         formula.XDisplayUnit = baseGrid.Dimension.Unit(baseGrid.DataInfo.DisplayUnitName);
+         formula.YDisplayUnit = valueColumn.Dimension.Unit(valueColumn.DataInfo.DisplayUnitName);
+
+         foreach (var timeValue in baseGrid.Values)
+         {
+            formula.AddPoint(timeValue, valueColumn.GetValue(timeValue).ToDouble());
+         }
+
+         return formula;
+      }
+
+      private TableFormula newTableFormula()
+      {
+         var tableFormula = _context.Create<TableFormula>();
+         tableFormula.XDimension = _context.DimensionFactory.Dimension(TIME);
+         tableFormula.XName = TIME;
+         tableFormula.Dimension = _context.DimensionFactory.Dimension(AppConstants.DimensionNames.FRACTION);
```

Comment:
Why do we assume here that the X dimension is always Time and Y dimension is always fraction?
Isn't it better to set the X dimension to `baseGrid.Dimension` and Y dimension to `valueColumn.Dimension`?

And don't we need to set the formula name somewhere?

Reply:
> Why do we assume here that the X dimension is always Time and Y dimension is always fraction? Isn't it better to set the X dimension to `baseGrid.Dimension` and Y dimension to `valueColumn.Dimension`?
> 
> And don't we need to set the formula name somewhere?

This tableformula is actually going to be thrown away. The interface for this is borrowed from PK-Sim where the formula in the parameter is swapped. In MoBi, we aren't going to change the formula,  we are only going to clear the points from the existing formula and add these new ones.

So, maybe we could change this in Core so the task only needs to return a data repository, then in PK-Sim the TableFormula is created, where in MoBi, I can just use the points from the DataRepository.

Reply:
Refactored so that we don't needlessly create the TableFormula. Instead, we use the Imported DataRepository